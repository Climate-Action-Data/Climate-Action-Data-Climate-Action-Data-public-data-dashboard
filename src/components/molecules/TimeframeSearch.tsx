import { Timeframes } from "@/@types/Timeframe"
import { Flex, Text, Box, Divider, HStack, Tag, Button, TagLabel, TagCloseButton } from "@chakra-ui/react"
import { useState } from "react"

export const TimeframeSearch = (): React.JSX.Element => {
    const [timeframe, setTimeframe] = useState<string | undefined>(undefined)
    return (
        <Flex height={"40px"} alignItems={"center"}>
            <HStack spacing={4}>
                {Timeframes.map((timeFrameName) => (
                    <Tag
                        size={"md"}
                        key={timeFrameName}
                        borderRadius='full'
                        _hover={{ cursor: "pointer" }}
                        variant={timeFrameName === timeframe ? "solid" : "outline"}
                        colorScheme='gray'
                    >
                        <TagLabel onClick={() => setTimeframe(timeFrameName)}>
                            {timeFrameName}
                        </TagLabel>
                        {timeFrameName === timeframe &&
                            <TagCloseButton onClick={() => setTimeframe(undefined)} />
                        }
                    </Tag>
                ))}
            </HStack>
        </Flex>
    )
}