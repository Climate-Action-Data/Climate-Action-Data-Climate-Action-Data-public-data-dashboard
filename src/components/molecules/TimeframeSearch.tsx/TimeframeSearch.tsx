import { Timeframes } from "@/@types/Timeframe"
import { Flex, HStack, Tag, TagLabel, TagCloseButton } from "@chakra-ui/react"
import { useState } from "react"

export const TimeframeSearch = (): React.JSX.Element => {
    const [timeframe, setTimeframe] = useState<string | undefined>(undefined)
    return (
        <Flex height={"40px"} alignItems={"center"}>
            <HStack spacing={4}>
                {Timeframes.map((timeFrameName, idx) => (
                    <Tag
                        size={"md"}
                        key={timeFrameName}

                        borderRadius='full'
                        _hover={{ cursor: "pointer" }}
                        variant={timeFrameName === timeframe ? "solid" : "outline"}
                        colorScheme='gray'
                    >
                        <TagLabel data-testid={`button-timeframe-${idx}`} onClick={() => setTimeframe(timeFrameName)}>
                            {timeFrameName}
                        </TagLabel>
                        {timeFrameName === timeframe &&
                            <TagCloseButton data-testid={`button-timeframe-close`} onClick={() => setTimeframe(undefined)} />
                        }
                    </Tag>
                ))}
            </HStack>
        </Flex>
    )
}