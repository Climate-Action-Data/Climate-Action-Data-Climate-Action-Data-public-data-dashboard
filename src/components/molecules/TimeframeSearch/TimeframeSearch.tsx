import { TimeframesData } from '@/@types/Timeframe'
import { useActions, useAppState } from '@/overmind'
import { Flex, HStack, Tag, TagLabel, TagCloseButton } from '@chakra-ui/react'
import { useTranslation } from 'react-i18next'

export const TimeframeSearch = (): React.JSX.Element => {
  const { t } = useTranslation(`home`)
  const { carbonReduction } = useAppState().analytics
  const { setTimeframe } = useActions().analytics

  return (
    <Flex height={`40px`} alignItems={`center`}>
      <HStack spacing={4}>
        {Object.values(TimeframesData).map(
          (timeFrameName, idx) =>
            timeFrameName !== TimeframesData.MAX && (
              <Tag
                size={`md`}
                key={timeFrameName}
                borderRadius="full"
                _hover={{ cursor: `pointer` }}
                variant={timeFrameName === carbonReduction.carbonMapDataFilters.timeframe ? `solid` : `outline`}
                colorScheme="gray"
              >
                <TagLabel data-testid={`button-timeframe-${idx}`} onClick={() => setTimeframe(timeFrameName)}>
                  {t(`timeframes.${timeFrameName}`)}
                </TagLabel>
                {timeFrameName === carbonReduction.carbonMapDataFilters.timeframe && (
                  <TagCloseButton data-testid={`button-timeframe-close`} onClick={() => setTimeframe(TimeframesData.MAX)} />
                )}
              </Tag>
            ),
        )}
      </HStack>
    </Flex>
  )
}
