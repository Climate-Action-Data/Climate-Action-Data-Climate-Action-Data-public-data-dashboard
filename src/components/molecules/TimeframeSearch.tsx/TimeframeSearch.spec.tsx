import { render, screen } from '@testing-library/react'
import { TimeframeSearch } from './TimeframeSearch'
import userEvent from '@testing-library/user-event'

it(`renders correctly`, () => {
  const { container } = render(<TimeframeSearch />)
  expect(container).toMatchSnapshot()
})

it(`renders with click on timeframe`, async () => {
  const { container } = render(<TimeframeSearch />)
  await userEvent.click(screen.getByTestId(`button-timeframe-0`))
  expect(container).toMatchSnapshot()
})

it(`renders with click on going back`, async () => {
  const { container } = render(<TimeframeSearch />)
  await userEvent.click(screen.getByTestId(`button-timeframe-0`))
  await userEvent.click(screen.getByTestId(`button-timeframe-close`))
  expect(container).toMatchSnapshot()
})
