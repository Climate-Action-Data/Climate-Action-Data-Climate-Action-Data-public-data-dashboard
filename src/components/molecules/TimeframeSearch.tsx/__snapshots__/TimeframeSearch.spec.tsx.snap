// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-hajjsy"
  >
    <div
      class="chakra-stack css-1mlvmbj"
    >
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-0"
        >
          Max
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-1"
        >
          12m
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-2"
        >
          YTD
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-3"
        >
          1m
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`renders with click on going back 1`] = `
<div>
  <div
    class="css-hajjsy"
  >
    <div
      class="chakra-stack css-1mlvmbj"
    >
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-0"
        >
          Max
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-1"
        >
          12m
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-2"
        >
          YTD
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-3"
        >
          1m
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`renders with click on timeframe 1`] = `
<div>
  <div
    class="css-hajjsy"
  >
    <div
      class="chakra-stack css-1mlvmbj"
    >
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-0"
        >
          Max
        </span>
        <button
          aria-label="close"
          class="css-sorhh"
          data-testid="button-timeframe-close"
          type="button"
        >
          <svg
            class="chakra-icon css-1md6f3y"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
              fill="currentColor"
            />
          </svg>
        </button>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-1"
        >
          12m
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-2"
        >
          YTD
        </span>
      </span>
      <span
        class="css-ew0hq8"
      >
        <span
          class="css-zvlevn"
          data-testid="button-timeframe-3"
        >
          1m
        </span>
      </span>
    </div>
  </div>
</div>
`;
